packages/gatsby/src/redux/actions/public.js

```diff
@@ -471,6 +471,12 @@ actions.deleteNode = (options: any, plugin: Plugin, args: any) => {
   // Always get node from the store, as the node we get as an arg
   // might already have been deleted.
   const node = getNode(id)
+
+  if (!node) {
```

Comment:
This means some `node` existence checks below can be dropped (like line 483 and 514)

Reply:
Dropped extra `node` existance check in ed997fb7b3e59afa2dfe5fa99900b22858c58e33